pkg/runtime/reconciler.go

```diff
@@ -106,14 +107,20 @@ func (r *reconciler) reconcile(req ctrlrt.Request) error {
 
 	acctID := r.getOwnerAccountID(res)
 	region := r.getRegion(res)
+	roleARN, _ := r.cache.Accounts.GetAccountRoleARN(string(acctID))
```

Comment:
How about we hide the cache implementation further and create a `reconciler.getRoleARN()` function similar to the `getOwnerAccountID` and `getRegion` functions.

Reply:
+1 That's a good idea! Done